pakyow-core/lib/core/middleware/req_path_normalizer.rb

```diff
@@ -0,0 +1,24 @@
+module Pakyow
+  module Middleware
+    # Rack compatible middleware that normalize the path if contains '//',
+    # it replace '//' with '/' and issue a 301 redirect to the new path.
+    #
+    # @api public
+    class ReqPathNormalizer
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if env['PATH_INFO'].include? '//'
+          # normalize the path and issue 301 redirect
+          env['PATH_INFO'].gsub!('//', '/')
```

Comment:
You might not want to mutate the env like this. I can't say for sure it's problematic, but it seems scary to me somehow.